d2bs/kolbot/libs/bots/MFHelper.js

```diff
@@ -179,21 +187,30 @@ MainLoop:
 			if (command !== oldCommand) {
 				oldCommand = command;
 
-				if (command.indexOf("kill") > -1) {
-					print("ÿc4MFHelperÿc0: Kill");
+				// Finish on quit command
+				if (command.indexOf("quit") > -1) {
+					break MainLoop;
+				}
 
-					split = command.split("kill ")[1];
+				// common for all commands, take the leader's tp.
+				// do not care about portal area, if leader has sent a command, he also just has made a tp.
+				// if, at worst case, the portal leads to wrong area, the command will not be executed and helper will go back to town at next loop.
+				delay(me.ping);
+				Pather.usePortal(null, player.name)
+				while (!me.area) {
```

Comment:
this is not needed, usePortal already checks for area change, use the retry method as in the original version
also there is no reason not to check the area since the info is available

Reply:
usePortal creates delay if we pass in the area, this makes the helper wait some seconds not taking the tp because leader area is the previous on, not the current one where the portal has been made.

Reply:
investigate the source of the delay in usePortal, area check should not cause any significant delay

Reply:
That is not usPortal that makes delay.
This is player.area which returns the previous area when tp appears.
For example, leader runs pit. Teleport to pit level 1, area = x-1. Teleport to pit level 2, area = x. Make tp, send command. Helper receive command, check leader’s area. Here is the problem. Leader’s area returns x-1, not refreshed. `usePortal(x-1, leader)` will delay failing retrying to use portal because there is no portal to x-1, but x. Wait for the next helper loop pass where leader’s area now correctly returns x.

Reply:
By using ˋusePortal(null, leader)ˋ we don’t check failable area, and works better because the command is sent just after making a tp, so the tp is obviously the right area. And we check area ˋif (me.area == leader.area)ˋ after taking portal so leader.area has time to be refreshed, while taking tp.